docs/core/deploying/native-aot/interop.md

```diff
@@ -0,0 +1,82 @@
+
+While native code interop works similarly to non-AOT deployments, there are some specifics that differ when publishing as native AOT.
+
+## Direct P/Invoke calls
+
+The P/Invoke calls in AOT compiled binaries are bound lazily at runtime by default for better compatibility. The AOT compiler can be configured to generate direct calls for selected P/Invoke methods that are bound during startup by the dynamic loader that comes with the operating system. The unmanaged libraries and entry points referenced via direct calls always have to be available at runtime, otherwise the native binary fails to start.
+
+The benefits of direct PInvoke calls are:
+
+- Direct PInvoke calls have *better steady state performance*
+- Direct PInvoke calls make it possible to *link the unmanaged dependencies statically*
+
+The direct PInvoke generation can be configured using `<DirectPInvoke>` items in the project file. The item name can be either `modulename`, which enables direct calls for all module entry points, or `modulename!entrypointname`, which enables a direct call for the specific module and entry point only.
+
+`<DirectPInvokeList>filename</DirectPInvokeList>` items in the project file allow specifying a list of entry points in an external file. This is useful when the specification of direct PInvoke calls is long and it is not practical to specify them using individual `<DirectPInvoke>` items. The file can contain empty lines and comments starting with `#`.
+
+Examples:
+
+```xml
+<ItemGroup>
+  <!-- Generate direct PInvoke calls for everything in __Internal -->
+  <!-- This option replicates Mono AOT behavior that generates direct PInvoke calls for __Internal -->
+  <DirectPInvoke Include="__Internal" />
+  <!-- Generate direct PInvoke calls for everything in libc (also matches libc.so on Linux or libc.dylib on macOS) --> 
+  <DirectPInvoke Include="libc" />
+  <!-- Generate direct PInvoke calls for Sleep in kernel32 (also matches kernel32.dll on Windows) -->
+  <DirectPInvoke Include="kernel32!Sleep" />
+  <!-- Generate direct PInvoke for all APIs listed in DirectXAPIs.txt -->
+  <DirectPInvokeList Include="DirectXAPIs.txt" />
+</ItemGroup>
+```
+
+On Windows, native AOT uses a pre-populated list of direct P/Invoke methods that are available on all supported versions of Windows.
+
+> [!WARNING]
+> Because direct P/Invoke methods are resolved by the operating system dynamic loader and not by the native AOT runtime library, direct P/Invoke methods will not respect the <xref:System.Runtime.InteropServices.DefaultDllImportSearchPathsAttribute>. The library search order will follow the dynamic loader rules as defined by the operating system. Some operating systems and loaders offer ways to control dynamic loading through linker flags (such as `/DEPENDENTLOADFLAG` on Windows or `-rpath` on Linux). See details below on how to specify linker flags.
```

Comment:
Additionally, the dynamic lookup of Unicode (`W`) or ANSI (`A`) prefixes won't be checked, right? It might make sense to add that too or perhaps update the phrasing to include a statement that none of the dynamic lookup policy for P/Invokes will occur.

Perhaps rephrasing and adding a comment to the following doc? https://learn.microsoft.com/dotnet/core/dependency-loading/default-probing#unmanaged-native-library-probing

Reply:
> Additionally, the dynamic lookup of Unicode (W) or ANSI (A) prefixes won't be checked, right?

The W/A lookup is done at compile time in the set of DirectPInvokes specified by the project. For example, if your PInvoke is `GetEnvironmentVariables` with `CharSet.Unicode`, it is going to be matched with `kernel32!GetEnvironmentVariablesW` DirectPInvoke.

The relevant code is at https://github.com/dotnet/runtime/blob/2f18df4291febc514781ecf92593d9d00f3d8f57/src/coreclr/tools/aot/ILCompiler/ConfigurablePInvokePolicy.cs#L101